### ISSUE TITLE
Manipulation handler constraints not applied when set in code

### ISSUE THREAD

**Issue by provencher** [Microsoft/MixedRealityToolkit-Unity#4052]
## Describe the bug

When modifying the constraints of a manipulation handler via code, the default settings are still applied.

## To reproduce

1. Add a manipulation handler to the scene
2. At runtime, try and change the constraints.
3. The manipulation handler will be effectively unchanged.

## Expected behavior

The manipulation handler should adapt to the changed constraints.

## Your Setup (please complete the following information)

- Unity Version 2018.3.8f1
- MRTK Version v2 RC1

## Target Platform (please complete the following information)

- HoloLens 2

**Comment by chrisfromwork** [comment]
Assigning to @julenka and @yoyozilla to find an owner

**Comment by julenka** [comment]
Thanks for the issue! I'll see if I can take a look at it this week.

**Comment by provencher** [comment]
@julenka @Yoyozilla I've done some work tracking this down.

It seems to be a mix of two things.
Changing handle visibility in the bounding box for scaling does not restrict two handed manipulation in the manipulation handler. With that said, the controls do work as expected.

Second, the handle visibility property changes via code do not persist with the default box activation mode. I exposed those in a PR
https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4071

**Comment by julenka** [comment]
Hi @provencher I'm a little confused about how the two things you are mentioning (changing handle visibility and persisting property changes) are related to the issue.

As I understand it, this bug is saying: "if i change manipulationhandler to rotate only about Y axis in code as opposed to asset, the constraint is not actually applied". 

So in other words if I do something like:

```
var mh = GetComponent<ManipulationHandler>();
mh.Rotation.ConstraintOnRotation = XAxisOnly
```

and then try to rotate an object, the object still rotates about all axes.

**Comment by provencher** [comment]
@julenka it's a weird bug.

So the jist is that I was setting which handles the bounding box would show, as your code example pointed out, but when it came time to hover over the box, no handles would show up at all.

I looked through the code, and this is because the focus change function in the bounding box returns before changing handle visibility if the enum I exposed was set to manual, which is the default value. This early return prevents the handles that should be visible from showing, as the code that determines visibility does a check against the show wireframe property, which is set on the focus change function.

Perhaps this exposes a deeper bug, but I didn't want to change core functionality of the box as that could have far reaching consequences. Simply changing the enum I exposed stops the focus change function from returning early, and the wireframe bool is properly set.

Part of the issue with my bug report is that I conflated manipulation constraints with bounding box handles thinking they were more related.

Perhaps constraints on the manipulation handler should affect which handles the bounding box shows, as the handles permit transformations the manipulation handler might forbid.
